crates/kit/src/libvirt/status.rs

```
@@ -0,0 +1,155 @@
+//! libvirt status command - show libvirt environment information
+//!
+//! This module provides a status command that outputs JSON metadata about
+//! the libvirt environment, including version information and domain count.
+
+use clap::Parser;
+use color_eyre::{eyre::Context, Result};
+use serde::{Deserialize, Serialize};
+use std::process::Command;
+
+use crate::domain_list::DomainLister;
+
+/// Options for the libvirt status command
+#[derive(Debug, Parser)]
+pub struct LibvirtStatusOpts {
+    /// Output format (yaml or json)
+    #[clap(long, default_value = "yaml", value_enum)]
+    pub format: OutputFormat,
+}
+
+/// Output format for status command
+#[derive(Debug, Clone, clap::ValueEnum)]
+pub enum OutputFormat {
+    /// YAML format (default, human-readable)
+    Yaml,
+    /// JSON format (machine-readable)
+    Json,
+}
+
+/// libvirt version information
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LibvirtVersion {
+    pub major: u32,
+    pub minor: u32,
+    pub micro: u32,
+    pub full_version: String,
+}
+
+/// libvirt status information
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LibvirtStatus {
+    pub version: Option<LibvirtVersion>,
+    pub supports_readonly_virtiofs: bool,
+    pub domain_count: usize,
+    pub running_domain_count: usize,
+}
+
+/// Parse libvirt version from virsh version output
+fn parse_libvirt_version() -> Result<Option<LibvirtVersion>> {
+    let output = Command::new("virsh")
+        .args(&["version"])
+        .output()
+        .with_context(|| "Failed to check libvirt version")?;
+
+    if !output.status.success() {
+        return Ok(None);
+    }
+
+    let version_output = String::from_utf8_lossy(&output.stdout);
```

Don't use `utf8_lossy` we should error out on invalid UTF-8 here.